Added logging behavior to silence widget logging

lbustelo opened the pull request:
Fixes #349  

This PR implements a behavior to encapsulate logging support. It exposes a `log` property that can be used to turn on individual levels of logging. The default is to be silent.

Also added reading `Urth.debug` from the global to turn on full logging from one place.

Commit:
Added logging behavior to silence widget logging
(c) Copyright IBM Corp. 2016

lbustelo commented:
Still has WIP because I need to put some units for this behavior. I also would like to get a second opinion.

@deedubbu @peller @aluu317

deedubbu commented on elements/urth-viz-table/urth-viz-table.html:
```diff
@@ -235,7 +235,7 @@
                 // in case user provides missing column headers
                 var datarows = this.datarows;
-                console.assert(datarows);
+                this._assert(datarows);
```
Where is `_assert` defined?

lbustelo replied:
ha! Thanks... this was giving me an error in testing... I did a bulk find/replace.

lbustelo replied:
BTW... why are we using assert? Its say `Non-standard` [here](https://developer.mozilla.org/en-US/docs/Web/API/console/assert)

deedubbu commented on elements/urth-core-behaviors/logging-behavior.html:
```diff
@@ -0,0 +1,103 @@
+              this._info.apply(this, arguments);
+            },
+
+            _info: function() {
```
My only hesitation is with using an underscore prefix for what is actually the public API for this behavior. I'm assuming underscore is used because some of the API names could be common in other elements and something like `logInfo` or `infoLog` is verbose/confusing.

At a minimum should provide some minimal documentation for these API, even though they'll show up as Private.

lbustelo replied:
I'll document for sure... I think the `_` is appropriate since these are not meant to be external api for an element, rather internal utility type methods.

deedubbu commented on elements/urth-core-behaviors/logging-behavior.html:
```diff
@@ -0,0 +1,103 @@
+        var Urth = window.Urth = window.Urth || {};
+
+        /**
+         * Behavior that ability to control the output of logs. It redefines the logging methods in Polymer.Base to
```
`Behavior that ability...` === `Behavior with ability...` maybe?

lbustelo commented:
Fixed tests and address the documentation concerns.

Also, added unit tests to the logging behavior. If travis is happy, this should be good to merge.

deedubbu commented:
👍
